File: Difficulty: Easy/Sort 0s, 1s and 2s/sort-0s-1s-and-2s.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    void swap(int *a, int *b) {
        int temp = *a;
        *a = *b;
        *b = temp;
    }
    
    void sort012(vector<int>& arr) {
        int n = arr.size();
        int left = 0, mid = 0, right = n - 1;
        
        while(mid <= right) {
            if(arr[mid] == 0) {
                swap(&arr[left], &arr[mid]);
                left++;
                mid++;
            } else if(arr[mid] == 1) {
                mid++;
            } else {
                swap(&arr[mid], &arr[right]);
                right--;
            }
        }
    }
};

//{ Driver Code Starts.
int main() {

    int t;
    cin >> t;
    cin.ignore();
    while (t--) {

        vector<int> a;
        string input;
        int num;

        getline(cin, input);
        stringstream s2(input);
        while (s2 >> num) {
            a.push_back(num);
        }
        Solution ob;
        ob.sort012(a);

        int n = a.size();
        for (int i = 0; i < n; i++) {
            cout << a[i] << " ";
        }

        cout << endl;
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Medium/Sort 0s, 1s and 2s/sort-0s-1s-and-2s.java
class Solution {
    public void sort012(int[] arr) {
        int l = 0;
        
        for(int i=0; i<arr.length; i++) {
            if(arr[i] == 0) {
                int temp = arr[i];
                arr[i] = arr[l];
                arr[l] = temp;
                l++;
            }
        }
       
        l = arr.length-1;
       
        for(int i=arr.length-1; i>=0; i--) {
            if(arr[i] == 2) {
                int temp = arr[i];
                arr[i] = arr[l];
                arr[l] = temp;
                l--;
            }
        }
    }
}